src/neuroconv/tools/nwb_helpers/_models/_zarr_dataset_models.py

```diff
@@ -0,0 +1,122 @@
+"""Base Pydantic models for the ZarrDatasetConfiguration."""
+from typing import Any, Dict, List, Literal, Union
+
+import numcodecs
+import zarr
+from pydantic import Field, root_validator
+
+from ._base_dataset_models import DatasetConfiguration
+
+_available_zarr_filters = (
```

Comment:
Your milleage might vary but I would define sets with names like

```
loosy_codes = {"astype", "bitround", "quantize"}   # 
```
And then substract them.

I find the nested structure with comments and initilaization in the fly hard to read. Variables are very cheap and with good naming work as documentation.

Reply:
Enhanced the readability here

Reply:
Thanks.